BitwardenShared/UI/Platform/Application/AppProcessor.swift

```diff
@@ -380,6 +381,25 @@ extension AppProcessor {
         }
     }
 
+    /// Sets the auth completion route if needed based on the rehydration state.
+    /// This is useful for when the timeout lock happens so after the user unlocks again,
+    /// it navigates to the previous view they were into (if it's one of the views we save).
+    private func restoreAuthCompletionRouteIfNeeded() async {
+        guard appExtensionDelegate?.isInAppExtension != true else {
+            return
+        }
+        do {
+            guard let target = try await services.rehydrationHelper.getSavedRehydratableTarget() else {
+                return
+            }
+
+            await coordinator?.handleEvent(.setAuthCompletionRoute(target.appRoute))
+            try await services.rehydrationHelper.clearAppRehydrationState()
```

Comment:
❓ Is this needed if you also have logic in the router to set the auth completion route?

Reply:
Nop, just forgot to remove this part after I implemented the router logic; thanks for caching this!